### PR TITLE
issue-212 : Split ApplicationPatternDocumentation.md to sub-topics

### LINKED ISSUE #212: substructuring the documentation of the Implementation of the ApplicationPattern
- [ ] Go through the content in the page, https://github.com/openBackhaul/ApplicationPattern/blob/develop/doc/ImplementingApplications/ApplicationPatternDocumentation/ApplicationPatternDocumentation.md
- [ ] Decompose the content of the document according to available sub-topics
- [ ] For each sub-topic,  create a separate .md file
- [ ] Create a parent .md file that provides the overview and the list of sub-topics.
- [ ] Link the sub-topics in the parent page to the corresponding .md files

### PR DESCRIPTION
**Pull request by DanaSunal**
Fixes #212

### CODE REVIEW

**Comment by DanaSunal** [comment]
Reviewer is @IswaryaaS

File: doc/ImplementingApplications/ConceptOfApplicationPatternPackage/ConceptOfApplicationPatternPackage.md [hunk 1]
```diff
@@ -14,7 +14,13 @@ This library provides APIs,
```
Comment (vanithavalluripalli9) [inline doc/ImplementingApplications/ConceptOfApplicationPatternPackage/ConceptOfApplicationPatternPackage.md:14]: can we update the applicationPatternPackages.png according to the latest version

**Review by vanithavalluripalli9 — CHANGES_REQUESTED** [review]

File: doc/ImplementingApplications/ConceptOfApplicationPatternPackage/ConceptOfApplicationPatternPackage.md [hunk 2]
```diff
@@ -14,7 +14,13 @@ This library provides APIs,
 
 ## Application Pattern Modules
 
-ApplicationPattern modules consists of the following sub-modules
+ApplicationPattern modules consists of the following sub-modules:
+
+### Commons
+This module consists of classes that defines the ONFModel core model.
```
[the remark this answers is not in the document]
Comment (DanaSunal) [reply doc/ImplementingApplications/ConceptOfApplicationPatternPackage/ConceptOfApplicationPatternPackage.md]: Done.

File: doc/ImplementingApplications/ConceptOfApplicationPatternPackage/ConceptOfApplicationPatternPackage.md (hunk 1 above)
Comment (DanaSunal) [reply doc/ImplementingApplications/ConceptOfApplicationPatternPackage/ConceptOfApplicationPatternPackage.md:14]: Done.

File: doc/ImplementingApplications/ConceptOfApplicationPatternPackage/ConceptOfApplicationPatternPackage.md [hunk 3]
```diff
@@ -14,44 +14,42 @@ This library provides APIs,
 
 ## Application Pattern Modules
 
-ApplicationPattern modules consists of the following sub-modules
```
Comment (vanithavalluripalli9) [inline doc/ImplementingApplications/ConceptOfApplicationPatternPackage/ConceptOfApplicationPatternPackage.md:17]: Lines 1 to 14 are overlapping with content from ApplicationPatternDcoumentation . We can modify the documentation as below : 

1) remove the lines from 1 to 14 in this document.
2) rename the document to ApplictionPatternModulesOverview

**Review by vanithavalluripalli9 — CHANGES_REQUESTED** [review]

File: doc/ImplementingApplications/ConceptOfApplicationPatternPackage/SubModulesOfRest/client.md [hunk 4]
```diff
@@ -0,0 +1,73 @@
...
+![colorline_blue](https://user-images.githubusercontent.com/57349523/154715704-2e1a7c51-17c2-47af-a46a-85bd613f4a53.jpg)
+
+**RequestBuilder.js**
+
+This module provides functionality to construct a REST request.
+
+*Function*:
+|**Method and description**|**Input parameters**|**Return type**|
+|---|---|---|
+|**BuildAndTriggerRESTRequest** <br>This function triggers a REST request by calling the RESTClient |{string} **remoteIpAddressAndPort** ip address, port of the client application in the format <ipaddress>:<port>. <br>{string} **operationName** service that needs to be addressed in the client application. <br>{string} **method** HTTP method for the REST request. <br>{string} **requestHeader** HTTP request header for the REST call. <br>{string} **requestBody** request body for the REST call. |{promise} returns the HTTP response received.|
```
Comment (vanithavalluripalli9) [inline doc/ImplementingApplications/ConceptOfApplicationPatternPackage/SubModulesOfRest/client.md:44]: remoteIpAddressAndPort parameter should be removed . Operation client uuid parameter to be added

**Review by vanithavalluripalli9 — CHANGES_REQUESTED** [review]

**Comment by vanithavalluripalli9** [comment]
Assigning the PR to @PrathibaJee

File: doc/ImplementingApplications/ApplicationPatternDocumentation/ApplicationPatternDocumentation.md
Comment (PrathibaJee) [inline doc/ImplementingApplications/ApplicationPatternDocumentation/ApplicationPatternDocumentation.md:1]: In line no 25 , change the link from 
(../../SpecifyingApplications/ForwardingList/ForwardingList.md)  to
(../../SpecifyingApplications/ConceptsOfForwardingList/ConceptsOfForwardingList.md)

**Review by PrathibaJee — CHANGES_REQUESTED** [review]
spliting the Application documentation looks fine. Requested for few minor changes. 
Also created a separate issue to update the onfModel files as per the latest changes and to add content to the file ConceptOfApplicationPatternPackage/SubModulesOfCommon/AppCommons.md https://github.com/openBackhaul/ApplicationPattern/issues/653

File: server/applicationPattern/applicationPattern/rest/server/ResponseHeader.js [hunk 5]
```diff
@@ -31,7 +29,6 @@ class ResponseHeader {
             this.xCorrelator = ResponseHeader.xCorrelatorGenerator();
         }
         this.execTime = ResponseHeader.executionTimeInMilliseconds(startTime);
-        this.backendTime = this.execTime;
```
Comment (PrathibaJee) [inline server/applicationPattern/applicationPattern/rest/server/ResponseHeader.js:34]: Kindly please let me know why the backendTime is getting removed from the ResponseHeader ?  Many thanks. 
Additionally, remove this change from this pull request , create a separate pull request for this change if necessary, as the context of the attached issue is not related to this change.

File: doc/ImplementingApplications/ConceptOfApplicationPatternPackage/SubModulesOfServices/ElasticsearchService.md [hunk 6]
```diff
@@ -0,0 +1,161 @@
+**ElasticsearchService.js**
+
+This is a description of file ElasticsearchService.js. For more information about the Elasticsearch concept, consult [here](../../ConceptOfElasticsearch/ConceptOfElasticsearch.md).
+
+* ElasticsearchService
+    * [new ElasticsearchService()](#new_ElasticsearchService_new)
```
[the remark this answers is not in the document]
Comment (DanaSunal) [reply doc/ImplementingApplications/ConceptOfApplicationPatternPackage/SubModulesOfServices/ElasticsearchService.md:6]: Done

File: doc/ImplementingApplications/ConceptOfApplicationPatternPackage/SubModulesOfRest/server.md [hunk 7]
```diff
@@ -0,0 +1,96 @@
+#### **Server**
+The modules and classes in this package provide functionality to support the REST server to formulate the HTTP response (header, body, response code) as per the requirement of the "ApplicationPattern specification".
+
+The rest module description contains the following content:
+
+* [Server](#server)
```
[the remark this answers is not in the document]
Comment (DanaSunal) [reply doc/ImplementingApplications/ConceptOfApplicationPatternPackage/SubModulesOfRest/server.md:6]: Done.

File: doc/ImplementingApplications/ConceptOfApplicationPatternPackage/SubModulesOfRest/client.md [hunk 8]
```diff
@@ -0,0 +1,73 @@
+#### **Client**
+The module in this package provides a REST client which is used to communicate with the REST server in other applications.
+|![apd_image14](https://user-images.githubusercontent.com/57349523/153877957-edfc9d5c-89cb-4af8-9931-e154d5725c99.png)|
+|---|
+
+The rest module description contains the following content:
+
+* [Client](#client)
```
[the remark this answers is not in the document]
Comment (DanaSunal) [reply doc/ImplementingApplications/ConceptOfApplicationPatternPackage/SubModulesOfRest/client.md:8]: Done.

File: doc/ImplementingApplications/ConceptOfApplicationPatternPackage/Images/ApplicationPatternPackages.puml [hunk 9]
```diff
@@ -0,0 +1,41 @@
...
+    [ResponseHeader]
+    [ResponseCode]
+
+    folder "responseBody" {
+      [ResponseValue]
+      [ConsequentAction]
+    }
+  }
+ }
+ frame "onfModel" {
```
[the remark this answers is not in the document]
Comment (DanaSunal) [reply doc/ImplementingApplications/ConceptOfApplicationPatternPackage/Images/ApplicationPatternPackages.puml:35]: I started with that, but the picture would be too granular. Onf model should have a separate picture.

File: doc/ImplementingApplications/ConceptOfApplicationPatternPackage/ConceptOfApplicationPatternPackage.md (hunk 3 above)
Comment (DanaSunal) [reply doc/ImplementingApplications/ConceptOfApplicationPatternPackage/ConceptOfApplicationPatternPackage.md:17]: Moved the lines from this file to ApplicationPatternDocumentation (the ones in that document were outdated).

File: doc/ImplementingApplications/ApplicationPatternDocumentation/ApplicationPatternDocumentation.md
Comment (DanaSunal) [reply doc/ImplementingApplications/ApplicationPatternDocumentation/ApplicationPatternDocumentation.md:1]: Done.

**Review by PrathibaJee — DISMISSED** [review]

**Review by PrathibaJee — APPROVED** [review]

**Pull request merged**